data_processing/eia.py

```diff
@@ -0,0 +1,1526 @@
+        'SC': 'Coal-based Synfuel',
+        'SGC': 'Coal-Derived Synthesis Gas',
+        'SGP': 'Synthesis Gas from Petroleum Coke',
+        'SLW': 'Sludge Waste',
+        'SUB': 'Subbituminous Coal',
+        'SUN': 'Solar',
+        'TDF': 'Tire-Derived Fuels',
+        'WAT': 'Water',
+        'WC': 'Waste Coal',
+        'WDL': 'Wood Waste Liquids',
+        'WDS': 'Wood Waste Solids',
+        'WH': 'Waste Heat',
+        'WND': 'Wind',
+        'WO': 'Waste Oil'
+    }
+
+    return rosetta_stone
+
+
+def mover_database():
+    """
+    This function generates a hard-coded dictionary that
+    translates the "Reported Prime Mover" code.
+    Returns:
+    --------
+    mover: dict
+        A dictionary containing the mover code as the key
+        and the translation as the value.
+    """
+
+    mover = {
+        'BA': 'Battery',
+        'BT': 'Binary Cycle Turbine',
+        'CA': 'Combined-Cycle Steam',
+        'CE': 'Compressed Air',
+        'CP': 'Concentrated Solar Power',
+        'CS': 'Combined-Cycle Steam and Combustion',
+        'CT': 'Combined-Cycle Combustion',
+        'ES': 'Energy Storage',
+        'FC': 'Fuel Cell',
+        'FW': 'Flywheel',
+        'GT': 'Gas Turbine',
+        'HA': 'Hydrokinetic Axial Flow Turbine',
+        'HB': 'Hydrokinetic Wave Buoy',
+        'HK': 'Hydrokinetic Other',
+        'HY': 'Hydraulic Turbine',
+        'IC': 'Internal Combustion Engine',
+        'PS': 'Pumped Storage',
+        'OT': 'Other',
+        'ST': 'Steam Turbine',
+        'PV': 'Photovoltaic',
+        'WT': 'Wind Turbine Onshore',
+        'WT': 'Wind Turbine Offshore'
+    }
+
+    return mover
+
+
+def generation(year: str):
+    '''
+    This function generates a nested dictionary of the Universities
+    present in the EIA spreadsheet and their generation amounts
+    by each fuel/mover type combination.
+
+    Parameters:
+    -----------
+    year: str
+        This is the string for the year being analyzed.
+
+    Returns:
+    --------
+    final: dict
+        Nested dictionary of the Universities present in the EIA
+        spreadsheet and their generation amounts in MWhr by each fuel/mover
+        type combination.
+
+    '''
+
+    if not isinstance(year, str):
+
+        raise TypeError(
+            "Argument 'year' must be of type 'str'."
+        )
+
+    allowed_years = [
+        '2008',
+        '2009',
+        '2010',
+        '2011',
+        '2012',
+        '2013',
+        '2014',
+        '2015',
+        '2016',
+        '2017',
+        '2018'
+    ]
+
+    if year not in allowed_years:
+
+        raise ValueError(
+            f"Not a valid year. Possibilities include: {*allowed_years,}."
+        )
+
+    if int(year) < 2011:
+
+        netGen = 'NET GENERATION (megawatthours)'
+
+        headers = 7
+
+    elif int(year) >= 2011:
+
+        netGen = 'Net Generation (Megawatthours)'
+
+        headers = 5
+
+    generation = pd.read_csv((path + year + '.csv'),
```

Comment:
```suggestion

    filename = fileprefix + year + ".csv"
    generation = pd.read_csv((path + filename),
```
This is better because it allows people to point to different folders where their data may be stored.